### ISSUE TITLE
[Badcase]: 在用带思考的数据集微调32B模型后推理时有时有思考有时没思考

### ISSUE THREAD

**Issue by EvilSwordman** [QwenLM/Qwen3#1438]
### Model Series

Qwen3

### What are the models used?

Qwen3-32B

### What is the scenario where the problem happened?

用LLaMA-Factory微调Qwen3-32B后，用vLLM部署推理，<think></think>内有时有思考有时没思考，并且大部分时候都是空的。

### Is this badcase known and can it be solved using avaiable techniques?

- [x] I have followed [the GitHub README](https://github.com/QwenLM/Qwen3).
- [x] I have checked [the Qwen documentation](https://qwen.readthedocs.io) and cannot find a solution there.
- [x] I have checked the documentation of the related framework and cannot find useful information.
- [x] I have searched [the issues](https://github.com/QwenLM/Qwen3/issues?q=is%3Aissue) and there is not a similar one.

### Information about environment

OS: Ubuntu 20.04
Python: Python 3.10
GPUs: 8 x NVIDIA A800
NVIDIA driver: 470.82.01
CUDA compiler: 11.4
PyTorch: 2.6.0+cu124

Package                        Version
---------------------------------------- -----------
vllm：                              0.8.5
llamafactory：               0.9.3.dev0

### Description

1. 这是微调的数据集的格式
```
{"instruction": "", "output": "<think>\n  thinking  \n</think>\n answer", "history": [], "prefix": "}
```
2. llamafactory的脚本

```
rm -rf /root/.cache/huggingface/datasets 
MASTER_PORT=$(shuf -n 1 -i 10000-65535)
deepspeed --include=localhost:0,1,2,3,4,5,6,7 --master_port $MASTER_PORT  src/train.py \
    --deepspeed examples/deepspeed/ds_z3_config.json \
    --stage sft \
    --model_name_or_path /mnt/data/pre_model/Qwen3-32B/ \
    --do_train \
    --dataset_dir  ‘dataset_dir’ \
    --dataset ‘dataset’ \
    --finetuning_type lora \
    --output_dir 'output_dir' \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 8 \
    --per_device_eval_batch_size 1 \
    --lr_scheduler_type cosine \
    --logging_steps 5 \
    --val_size 0.05 \
    --save_strategy epoch \
    --eval_strategy epoch \
    --num_train_epochs 6 \
    --plot_loss \
    --learning_rate 3e-5 \
    --bf16 True \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --load_best_model_at_end True \
    --cutoff_len 8192 \
    --disable_gradient_checkpointing False \
    --overwrite_cache True \
    --preprocessing_num_workers 64 \
    --overwrite_output_dir True
``` 

3. vllm的脚本

```
CUDA_VISIBLE_DEVICES=4,5,6,7 python -m vllm.entrypoints.openai.api_server \
    --model ‘model’ \
    --served-model-name Qwen3 \
    --host 0.0.0.0 \
    --port 8965 \
    --tensor-parallel-size 4 \
    --gpu-memory-utilization 0.9 \
    --dtype float16 \
    --trust-remote-code \
    --enforce-eager \
    --uvicorn-log-level warning
```

4.服务的代码

```
from openai import OpenAI

client = OpenAI(
        base_url="http://localhost:8965/v1",
        api_key="empty",
)

conversation_history = [
    {"role": "system", "content": '''
     #任务相关的prompts
'''
    }
]

print("Start chatting (type 'exit' to end the conversation):")

while True:
    user_input = input("User: ")
    user_input = user_input + ' /think'
    # 检查是否退出对话
    if user_input.lower() in ["exit", "quit"]:
        print("Ending the conversation. Goodbye!")
        break

    # 将用户输入添加到对话历史
    conversation_history.append({"role": "user", "content": user_input})

    # 调用模型生成响应
    chat_response = client.chat.completions.create(
        model="Qwen3",
        messages=conversation_history,
        temperature=0,
        top_p=1,
        #presence_penalty=1.5,
        max_tokens=512
    )

    # 获取模型回复内容
    assistant_response = chat_response.choices[0].message.content

    # 将模型回复添加到对话历史
    conversation_history.append({"role": "assistant", "content": assistant_response})

    # 打印模型回复
    print(f"Assistant: {assistant_response}")
    print("")
```

<img width="1075" alt="Image" src="https://github.com/user-attachments/assets/9083c380-d149-4e06-aa4e-b7f8f63a8580" />
并且大部分时候think内容都是空的。

**Comment by jklj077** [comment]
1. you may need to finetune the LoRA hyperparameters
2. make sure the loss of the assistant messages in history messages is masked
3. you can try the following data format    
    ```
    {"instruction": "", "output": "<think>\nthinking\n</think>\n\nanswer", "history": [], "prefix": "}
    ```
